src/inmanta/util/__init__.py

```diff
@@ -863,3 +868,66 @@ def check_for_pool_exhaustion(self) -> None:
 
     def _reset_counter(self) -> None:
         self._exhausted_pool_events_count = 0
+
+
+def remove_comment_part(to_clean: str) -> str:
+    """
+    Remove the comment part of a given string and ensure that the lenght of the string is greater than 0
+
+    :param to_clean: The string to clean
+    :return: A cleaned string
+    """
+    drop_comment, _, _ = to_clean.partition("#")
```

Comment:
This assumed a requires can't contain a '#' , it can (as ugly as it is)
https://peps.python.org/pep-0508/ 

to refer to the spec: https://pip.pypa.io/en/latest/reference/requirements-file-format/#comments

```
A line that begins with # is treated as a comment and ignored. Whitespace followed by a # causes the # and the remainder of the line to be treated as a comment.
```

Reply:
Indeed, I missed that a # could be part of a requirement name